internal/pod_models_test.go

```diff
@@ -3181,4 +3181,114 @@ var _ = Describe("pod_models", func() {
 			Expect(GetPodDNSName(cluster, "operator-test-storage-1")).To(Equal("operator-test-storage-1.operator-test-1.my-ns.svc.cluster.example"))
 		})
 	})
+
+	DescribeTable("getting the sidecar image from the pod spec", func(spec *corev1.PodSpec, expected string) {
+		Expect(GetSidecarImageFromPodSpec(spec)).To(Equal(expected))
+	},
+		Entry("sidecar container exists",
+			&corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  MainContainer,
+						Image: "main",
+					},
+					{
+						Name:  SidecarContainer,
+						Image: "sidecar",
+					},
+				},
+			}, "sidecar"),
+		Entry("sidecar container is absent",
+			&corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  MainContainer,
+						Image: "main",
+					},
+				},
+			}, ""),
+	)
+
+	When("checking if the pod spec matches the expected spec", func() {
+		var cluster *fdbv1beta2.FoundationDBCluster
+		var id string
+		var pod corev1.Pod
+
+		When("the cluster is upgraded", func() {
+			BeforeEach(func() {
+				cluster = CreateDefaultCluster()
+				err = NormalizeClusterSpec(cluster, DeprecationOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				cluster.Status.RunningVersion = cluster.Spec.Version
+				cluster.Spec.Version = fdbv1beta2.Versions.NextMajorVersion.String()
+				id = "storage-1"
+			})
+
+			When("the sidecar matches", func() {
+				BeforeEach(func() {
+					pod = corev1.Pod{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  MainContainer,
+									Image: "main",
+								},
+								{
+									Name:  SidecarContainer,
+									Image: fmt.Sprintf("%s:%s-1", SidecarDefaultImage, cluster.Spec.Version),
+								},
+							},
+						},
+					}
+				})
+
+				It("should return that the spec matches", func() {
+					Expect(cluster.IsBeingUpgraded()).To(BeTrue())
+					correctSpec, err := PodHasCorrectSpec(cluster, fdbv1beta2.ProcessClassStorage, id, pod)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(correctSpec).To(BeTrue())
+				})
+			})
+
+			When("the sidecar doesn't matches", func() {
+				BeforeEach(func() {
+					pod = corev1.Pod{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  MainContainer,
+									Image: "main",
+								},
+								{
+									Name:  SidecarContainer,
+									Image: fmt.Sprintf("%s:%s-1", SidecarDefaultImage, cluster.Status.RunningVersion),
+								},
+							},
+						},
+					}
+				})
+
+				It("should return that the spec doesn't match", func() {
+					Expect(cluster.IsBeingUpgraded()).To(BeTrue())
+					correctSpec, err := PodHasCorrectSpec(cluster, fdbv1beta2.ProcessClassStorage, id, pod)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(correctSpec).To(BeFalse())
+				})
+			})
+
+		})
+	})
 })
+
+/*
+
+add tests:
```

Comment:
Does this need additional action, or should we remove it?

Reply:
Oh, I forgot to remove that.